api/src/main/java/org/apache/iceberg/types/Types.java

```diff
@@ -524,6 +563,91 @@ public boolean equals(Object o) {
     public int hashCode() {
       return Objects.hash(NestedField.class, id, isOptional, name, type);
     }
+
+    private static JsonNode validateDefault(String name, Type type, JsonNode defaultValue) {
```

Comment:
I think this logic should be moved to a `ValueParser` with `toJson` and `fromJson` methods, like the others. We don't want to leak any Jackson classes in the API.